release-23.1: status: add network packet dropped and error metrics

nvb opened the pull request:
Backport 1/2 commits from #116005.

/cc @cockroachdb/release

Release justification: improves observability of networking instability. Assists L2 escalations.

---

This commit adds packet drop/error counter on network's sending and receiving end. These metrics are derived from the os and aims for better observability on tail latencies.

The second added chart looks like below for number of packets, packets error, and packets dropped on sending and receiving end.

<img width="1055" alt="Screenshot 2023-12-13 at 4 49 11 PM" src="https://github.com/cockroachdb/cockroach/assets/20375035/20cfc33c-9db6-4492-a9d6-9194454ef360">





Fixes: #115458

Release note(ui change): the DB Console metrics dashboard's Networking tab is
enhanced with charts that visualize number of packets received, number of
receiving packets with error, number of receiving packets got dropped, number
of packets sent, number of sending packets with error, and number of sending
packets got dropped.

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/116727)
<!-- Reviewable:end -->

lyang24 commented:
Seems like v23.1 is a bit different where we define the charts [here](https://github.com/cockroachdb/cockroach/blob/release-23.1/pkg/ts/catalog/chart_catalog.go#L141C1-L159C4) and removed on later versions. Should i start a new pr for 23.1 or its better to skip v23.1?

dhartunian commented:
@lyang24 can you add the charts to the chart catalog by amending this commit? the chart catalog was removed because it was unused so it doesn't really matter where you place them in there as long as they satisfy CI.

lyang24 committed:
status: add network packet dropped and error metrics

This commit adds packet drop/error counter on network's sending and receiving
end. These metrics are derived from the os and aim for improving observability
on tail latencies.

Fixes: #115458

Release note: None

dhartunian commented:
@nvanbenschoten did you intentionally omit the DB Console metrics page changes in the backport? They're in the second commit on the original PR.

lyang24 commented:
> @nvanbenschoten did you intentionally omit the DB Console metrics page changes in the backport? They're in the second commit on the original PR.

Could it be the network metrics is mixed with [hardware metrics](https://github.com/cockroachdb/cockroach/blob/80ab3499ff6a4fc9d9b8d55008784aeb0615d5b0/pkg/ui/workspaces/db-console/src/views/cluster/containers/nodeGraphs/dashboards/hardware.tsx#L165C1-L190C5) on v23.1 and later got separated to its own metrics? does is make sense for me to update this page as well?

dhartunian commented:
@nvanbenschoten ah makes sense. yeah maybe throw them into `hardware.tsx` for 23.1

lyang24 commented:
<img width="1193" alt="image" src="https://github.com/cockroachdb/cockroach/assets/20375035/d4cea312-564b-433c-8bd2-398a81813064">

Hi, i have added the charts from to the hardware graphs. Its working on cockroach demo. please node the subtle difference that tenent source is not available and it does not appear in other charts so i removed it.
removed in chart definition:
`tenantSource={tenantSource`}

dhartunian reviewed (approved):
LGTM, thanks for the changes

lyang24 commented:
test is not happy with `showMetricsInTooltip` and other charts do not have this so i replaced with `sources={nodeSources}` to be consistent with other charts
```
    src/views/cluster/containers/nodeGraphs/dashboards/hardware.tsx:265:53 - error TS2322: Type '{ children: Element; title: string; showMetricsInTooltip: boolean; }' is not assignable to type 'IntrinsicAttributes & OwnProps & { children?: ReactNode; }'.
      Property 'showMetricsInTooltip' does not exist on type 'IntrinsicAttributes & OwnProps & { children?: ReactNode; }'.

    265     <LineGraph title="Network Packet Drops on Send" showMetricsInTooltip={true}>
```

lyang24 commented:
Hi @dhartunian , thank you for guiding through iterations all tests passed. Do you mind take one more look when you have time?

lyang24 commented:
Hi @nvanbenschoten, do you want to take a look on this when you have time? we are planning to do a minor version upgrade at the beginning of the year I hope we can include this back port.

nvb reviewed (commented):
Thanks @lyang24 for updating this with the DB console changes. They <img class="emoji" title=":lgtm:" alt=":lgtm:" align="absmiddle" src="https://reviewable.io/lgtm.png" height="20" width="61"/>.

> Reviewed 4 of 4 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/cockroachdb/cockroach/116727#-:-Nnk1WWkDWEBBD76Jx5-:bvaq67z)* status: <span>:shipit:</span> complete! 1 of 0 LGTMs obtained (waiting on @dhartunian)



<!-- Sent from Reviewable.io -->
